app/views/handlers/list_add_question.py

```diff
@@ -12,5 +12,8 @@ def handle_post(self, form):
             form, self.parent_block['for_list']
         )
         self.questionnaire_store_updater.update_answers(form)
+        self._questionnaire_store_updater.remove_completed_relationship_locations_for_list_name(
```

Comment:
This should be done as part of the `add_list_item_and_answers` method as:
- This should always be done when adding list items.
- We should keep relationship functionality in questionnaire store updater.